contracts/src/Token.sol

```diff
@@ -28,33 +28,29 @@ import {ERC20Lib} from "./ERC20Lib.sol";
  * functions have been added to mitigate the well-known issues around setting
  * allowances. See {IERC20-approve}.
  */
-contract ERC20 is IERC20, IERC20Permit {
-    using ERC20Lib for ERC20Lib.TokenStorage;
+contract Token is IERC20, IERC20Permit {
+    using TokenLib for TokenLib.Token;
 
-    error Unauthorized();
-
-    ERC20Lib.TokenStorage token;
-
-    address public immutable OWNER;
-
-    uint8 public immutable decimals;
+    TokenLib.Token token;
```

Comment:
Would suggest to move this dynamic slots to the very end.

Reply:
Moved fields around a bit